python/tvm/relay/frontend/tensorflow.py

```diff
@@ -1319,8 +1420,53 @@ def from_tensorflow(self, graph, layout="NHWC", shape=None, outputs=None):
                         inputs.append(self._nodes[i][0])
                         input_shapes[self._nodes[i][0]] = self._output_shapes[i]
                 attr['_input_shapes'] = input_shapes
+                node_name_prefix = node.name.rsplit('/', 1)[0]
 
-                op = self._convert_operator(node.op, inputs, attr, graph)
+                if node.op == "Merge":
+                    if _in_while_loop(control_flow_node_map, node_name_prefix):
```

Comment:
does `_in_while_loop` depend on specific node traversal order(topological order of nodes) to work correctly? If yes please add a comment.

Reply:
No, it doesn't, since the graph has been traversed to get control_flow_node_map dict before this traversal, every node needed is already added in the control_flow_node_map.

Reply:
thanks. I expand the diff and now I see the loop before this loop.